src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -544,16 +544,18 @@ private bool RequiresEncoding(StringSegment input)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int GetBase64Length(int inputLength)
         {
-            // Copied from https://github.com/dotnet/runtime/blob/82ca681cbac89d813a3ce397e0c665e6c051ed67/src/libraries/System.Private.CoreLib/src/System/Convert.cs#L2530
-            long outlen = ((long)inputLength) / 3 * 4;          // the base length - we want integer division here.
-            outlen += ((inputLength % 3) != 0) ? 4 : 0;         // at most 4 more chars for the remainder
+            // Based on https://github.com/dotnet/runtime/blob/ac21254184f5a6cec884d8d4049c80d905170b38/src/libraries/System.Memory/src/System/Buffers/Text/Base64Encoder.cs#L150
 
-            if (outlen > int.MaxValue)
+            const int MaximumEncodeLength = (int.MaxValue / 4) * 3; // 1610612733
+
+            if ((uint)inputLength <= MaximumEncodeLength)
             {
-                throw new OutOfMemoryException();
+                return ((inputLength + 2) / 3) * 4;
```

Comment:
I feel like I'm missing something here. Isn't this the exact same as the method `Base64Encoder.GetMaxEncodedToUtf8Length`? And I thought that method gave you the max, not the exact length.

Reply:
> Isn't this the exact same as the method 

Yes, except different exception. I'm not sure which exception (OutOfMemory or ArgumentOutOfRange) is the better one -- I'd prefer the latter as used by Base64Encoder.GetMaxEncodedToUtf8Length.

The naming with "max" is IMHO not ideal for _encoding_, as the exact lenght can be computed -- as done by (l + 2) / 3 * 4.
For decoding the exact length can't be computed, due padding, so there "max" makes sense, and I assume the naming was choses for "symmetry".

Would you prefer to just forward to `Base64Encoder.GetMaxEncodedToUtf8Length` or use that method directly?

Reply:
OutOfMemoryException is pretty strange given an this method doesn't attempt any allocations. It only makes sense if the caller would immediately allocate for it. I'd recomend ArgumentOutOfRange too.

Reply:
OK, I'll use `Base64.GetMaxEncodedToUtf8Length` directly, as everything we need is there.

Reply:
Yeah that "Max" is pretty awful in the API and confused me, looks like just using the badly named API is best.